src/components/SwiperContainer.tsx

```diff
@@ -0,0 +1,72 @@
+import { Swiper, SwiperSlide } from 'swiper/react';
+
+import 'swiper/css';
+import { A11y, Navigation, Pagination, Scrollbar } from "swiper/modules";
+import Card from "./Card";
+import PopularResContent from "./PopularResContent";
+import SignatureDishContent from "./SignatureDishContent";
+
+const contentComponents: { [key: string]: React.ComponentType<any> } = {
+    popularRes: PopularResContent,
+    signatureDish: SignatureDishContent,
+
+}
+
+interface SwiperProps {
+    cards: {
+        title: string;
+        img: string;
+        content: {
+            type: string
+            chef?: string;
+            rating?: number;
+            icons?: IconProps[]
+            ingredients?: string[]
+            price?: number
+        }
+    }[]
+    spaceBetween: number
+    slidesPerView: number
+    mainTitle: string
+}
+
+export interface IconProps {
```

Comment:
best be in external type/interface file (usually will be in src->data folder -> types folder)